action/protocol/staking/handlers_test.go

```diff
@@ -166,6 +167,1624 @@ func TestProtocol_HandleCreateStake(t *testing.T) {
+		},
+		// ErrInvalidOwner
+		{
+			identityset.Address(1),
+			"10000000000000000000",
+			100,
+			false,
+			1,
+			"test2",
+			big.NewInt(unit.Qev),
+			10000,
+			1,
+			1,
+			time.Now(),
+			10000,
+			true,
+			ErrInvalidOwner,
+			iotextypes.ReceiptStatus_Success,
+		},
+		// Upsert error cannot happen,because CreateStake already check collision
+		// change from 0 to test1
+		{
+			identityset.Address(2),
+			"10000000000000000000",
+			100,
+			false,
+			0,
+			"test1",
+			big.NewInt(unit.Qev),
+			10000,
+			1,
+			1,
+			time.Now(),
+			10000,
+			false,
+			nil,
+			iotextypes.ReceiptStatus_Success,
+		},
+	}
+
+	for _, test := range tests {
+		sm, p, candidate, candidate2 := initAll(t, ctrl)
+		// candidate2 vote self,index 0
+		initCreateStake(t, sm, candidate2.Owner, 100, big.NewInt(unit.Qev), 10000, 1, 1, time.Now(), 10000, p, candidate2, "10000000000000000000", false)
+		// candidate vote self,index 1
+		_, createCost := initCreateStake(t, sm, candidate.Owner, test.initBalance, test.gasPrice, test.gasLimit, test.nonce, test.blkHeight, test.blkTimestamp, test.blkGasLimit, p, candidate, test.amount, false)
+
+		act, err := action.NewChangeCandidate(test.nonce, test.candidateName, test.index, nil, test.gasLimit, test.gasPrice)
+		require.NoError(err)
+		intrinsic, err := act.IntrinsicGas()
+		require.NoError(err)
+		ctx := protocol.WithActionCtx(context.Background(), protocol.ActionCtx{
+			Caller:       test.caller,
+			GasPrice:     test.gasPrice,
+			IntrinsicGas: intrinsic,
+			Nonce:        test.nonce,
+		})
+		ctx = protocol.WithBlockCtx(ctx, protocol.BlockCtx{
+			BlockHeight:    1,
+			BlockTimeStamp: time.Now(),
+			GasLimit:       1000000,
+		})
+		if test.clear {
+			cc := p.inMemCandidates.GetBySelfStakingIndex(test.index)
+			p.inMemCandidates.Delete(cc.Owner)
+		}
+		r, err := p.handleChangeCandidate(ctx, act, sm)
+		require.Equal(test.err, errors.Cause(err))
+		if r != nil {
+			require.Equal(uint64(test.status), r.Status)
+		} else {
+			require.Equal(test.status, iotextypes.ReceiptStatus_Success)
+		}
+
+		if test.err == nil && test.status == iotextypes.ReceiptStatus_Success {
+			// test bucket index and bucket
+			bucketIndices, err := getCandBucketIndices(sm, identityset.Address(1))
+			require.NoError(err)
+			require.Equal(2, len(*bucketIndices))
+			bucketIndices, err = getVoterBucketIndices(sm, identityset.Address(1))
+			require.NoError(err)
+			require.Equal(1, len(*bucketIndices))
+			indices := *bucketIndices
+			bucket, err := getBucket(sm, indices[0])
+			require.NoError(err)
+			require.Equal(identityset.Address(1).String(), bucket.Candidate.String())
+			require.Equal(identityset.Address(1).String(), bucket.Owner.String())
+			require.Equal(test.amount, bucket.StakedAmount.String())
+
+			// test candidate
+			candidate, err := getCandidate(sm, candidate.Owner)
+			require.NotNil(candidate)
+			require.NoError(err)
+			require.Equal("20000000000000000002", candidate.Votes.String())
```

Comment:
initBalance + xxx?

Reply:
votes is 2 when call initAll,I will change to 0 in next pr